bb8/tests/test.rs

```diff
@@ -104,6 +104,37 @@ where
     }
 }
 
+struct BrokenConnectionManager<C> {
+    _c: PhantomData<C>,
+}
+
+impl<C> BrokenConnectionManager<C> {
+    fn new() -> Self {
+        BrokenConnectionManager { _c: PhantomData }
+    }
+}
+
+#[async_trait]
```

Comment:
Sorry, missed this previously: I want to review this move separately, so please isolate it in a separate commit (and squash all the other commits) together.

Reply:
The idea here is that I want the move of the existing code here isolated in a separate commit from adding the new test -- sorry if that wasn't clear from my previous comment.

Reply:
I think I understand? Let me try again.

Reply:
Not quite, but I've taken the liberty of fixing it up.